haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -456,10 +456,72 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _CohereEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        # pretrained embedding models coming from:
+        self.max_seq_len = retriever.max_seq_len
```

Comment:
Should we overwrite `max_seq_len` here with the value that is allowed for cohere models? `max_seq_len` for cohere models is 512, but cohere API can handle up to 4096 tokens. Still, they recommend reducing text length to be under 512 tokens (see [here](https://docs.cohere.ai/embed-reference#texts)).

I think the same question would be applicable to the `_OpenAIEmbeddingEncoder`.

Reply:
Let's do max function on value just like we'll do for batch size.